src/main/kotlin/NaiveBayesBinaryClassifier.kt

```diff
@@ -4,72 +4,53 @@ class NaiveBayesBinaryClassifier {
     var logPrior: Double = 0.0
     var vocabulary = emptyMap<String, Double>()
 
-    private fun buildFrequences(texts: List<List<String>>, targets:List<Int>): Map<String, Pair<Int, Int>>{
+    private fun buildFrequencies(texts: List<List<String>>, targets: List<Int>): Map<String, Pair<Int, Int>> {
         // texts - list of tokenized tweets, targets = labels (will need to combine positive and negative tweets)
         // frequency table of word to Pair<negative (0) count , positive (1) count>
-        val frequencyTable = mutableMapOf<String, Pair<Int,Int>>()
-        for ((tweet, y)  in texts.zip(targets)) {
+        val frequencyTable = mutableMapOf<String, Pair<Int, Int>>()
+        for ((tweet, y) in texts.zip(targets)) {
             for (word in tweet) {
-                val counts = frequencyTable.getOrDefault(word, Pair(0,0))
-                if (y == 0) frequencyTable.put(word, Pair(counts.first + 1, counts.second))
-                if (y == 1) frequencyTable.put(word, Pair(counts.first, counts.second + 1))
+                val (negativeCount, positiveCount) = frequencyTable.getOrDefault(word, 0 to 0)
+                if (y == 0) frequencyTable.put(word, negativeCount + 1 to positiveCount)
+                if (y == 1) frequencyTable.put(word, negativeCount to positiveCount + 1)
             }
         }
         return frequencyTable
     }
 
     private fun computeLogLambdas(freqs: Map<String, Pair<Int, Int>>): Map<String, Double> {
         val allPositiveCounts = freqs.values.sumBy { it.second }
-        val allNegativeCounts = freqs.values.sumBy {it.first}
+        val allNegativeCounts = freqs.values.sumBy { it.first }
         val vocabLength = freqs.size
 
-        val logLamdas = mutableMapOf<String, Double>()
-
-        for (word in freqs.keys) {
-            // counting probabilities with Laplacian smoothing to avoid 0s
-            val posProb = ((freqs.getValue(word).second + 1).toDouble() / (allPositiveCounts + vocabLength))
-            val negProb = ((freqs.getValue(word).first + 1).toDouble() / (allNegativeCounts + vocabLength))
-            val logLambda = ln(posProb/negProb)
-            logLamdas[word] = logLambda
+        return freqs.keys.associate { word ->
```

Comment:
This is new to me. Took me a moment to figure out how it works, but looks interesting!